keylime/tpm/tpm_abstract.py

```diff
@@ -140,43 +149,50 @@ def _set_tpm_metadata(self, key, value):
             self.__write_tpm_data()
 
     @abstractmethod
-    def tpm_init(self, self_activate=False, config_pw=None):
+    def tpm_init(self, self_activate: bool = False, config_pw: Optional[str] = None):
         pass
 
     # tpm_quote
     @abstractmethod
-    def create_quote(self, nonce, data=None, pcrmask=EMPTYMASK, hash_alg=None, compress=False):
+    def create_quote(
+        self,
+        nonce: str,
+        data: Optional[bytes] = None,
+        pcrmask: str = EMPTYMASK,
+        hash_alg: Optional[str] = None,
+        compress: bool = False,
+    ):
         pass
 
     @abstractmethod
     def check_quote(
         self,
-        agentAttestState,
-        nonce,
+        agentAttestState: AgentAttestState,
+        nonce: str,
         data,
-        quote,
-        aikTpmFromRegistrar,
-        tpm_policy=None,
-        ima_measurement_list=None,
+        quote: str,
+        aikTpmFromRegistrar: str,
+        tpm_policy: Optional[Union[str, Dict]] = None,
+        ima_measurement_list: Optional[List[str]] = None,
         allowlist=None,
-        hash_alg=None,
-        ima_keyrings=None,
-        mb_measurement_list=None,
+        hash_alg: Optional[str] = None,
+        ima_keyrings: Optional[ImaKeyrings] = None,
+        mb_measurement_list: Optional[str] = None,  # FIXME
```

Comment:
This actually must not be None since the function that will be called with it unconditionally (so far) only accept `str`.

Reply:
Fixed the called function to accept `Optional[str]` which resolved this series of FIXMEs.